docker/create-env.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+# Copy the .dist environment files and wait them modifications.
+# Merge them to .env file for docker-compose.
+
+# Usage:
+# `./create-env.sh`
+# `./create-env.sh -q  # Quite mode`
+# `./create-env.sh --quite  # Quite mode`
+
+set -e
+
+QUITE=false
+for arg in "$@"
+do
+    if [[ $arg = "--quite" || $arg = "-q" ]]
+    then
+        QUITE=true
+    fi
+done
+
+function create_env_files {
+    for file in env_files/*.dist
+    do
+        new_file=${file%.dist}
+
+        if [[ ! -f "$new_file" ]]
+        then
+            cp $file $new_file
+        fi
+
+        if ! $QUITE
+        then
+            echo "Checkout the $new_file and configure it."
+            read -p "Are you ready to continue? [y/n]: " yn
+            if [[ $yn != "y" ]]
+            then
+                exit 1
+            fi
+        fi
+
+        cat $new_file >> .env
```

Comment:
wait ex dist files modification is not obvious and clear for use.
We should explain several steps throw bash script interface or throw doc.
Moreover `.env` file contains constants duplication

Only `build/docker-compose.yml` uses `.env` file, as i see.

We can just add smth like it to this docker-compose:
```
env_file:
 - env_files/app
 - env_files/credentials
 - env_files/paths
 - env_files/ports
```

... and rm `.env` file with it's additional bash dialog

Reply:
@duker33 
>wait ex dist files modification

We do not wait dist files modification, instead we copy dist files and wait modifications of these copies. `create-env.sh` already has doc for this process. 

>Only build/docker-compose.yml uses .env file, as i see.

The `.env` file is required to configure options of docker-compose files such as: `$SRC_DIR`, `$DEPS_DIR`, `$REDIS_PASSWORD`, etc.
The `env_file` option pass env variables only to container, but not to `docker-compose.yml`

Reply:
@artemiy312 , 

> We do not wait dist files modification, instead we copy dist files and wait modifications of these copies.

y, copies, but it does not matter for UX. I thought about it for a little. We have not simple UX decision only because of not simple task. So, not UX is complex, but designed good. 
Only one moment: i think we can copy all files the same time. And then ask user to edit them in a bunch.
He'll push only one "y" button instead of several

> The env_file option pass env variables only to container

got it. Another non-obvious docker-composer detail. One not good thing, that stayed: after `create-env.sh` launched we'll have `.env` file in repo root and `env_files/` too. It'll be confusing.
Let's name it `.env.build` and add this comment `Used only for images build. See env_files/ folder to change containers behaviour.`

Reply:
@duker33 
>And then ask user to edit them in a bunch.

I going to do it

>after create-env.sh launched we'll have .env file in repo root and env_files/ too. It'll be confusing.

I do not get the point. Currently,  the `create-env.sh` creates only one `.env` file in `docker/` directory, isn't it? Also we can not to name an environment file as `.env.build`, because docker-compose requires certain name for it - `.env`

Reply:
@artemiy312 , 

> Currently, the create-env.sh creates only one .env file in docker/ directory, isn't it?

but we also have `env_files/` dir with vars. It's not obvious, that .env is only for build, but env_files/ are only for containers. Every new proger (or we after half a year) will spend time on debugging and config digging trying to change env var for container

> Also we can not to name an environment file as .env.build

hmm, it's strange, `.env` file is visible for build, `env_file: .env.build` is not visible already.
If it's another docker-compose amazing detail, then we have to leave it as is. But we can add comment at top of this file

Reply:
tested it:

```
# docker/images/python/dev/Dockerfile
FROM python:slim

RUN echo $SRC_DIR && echo 'bugaga'
...
```

got this output:
```
andre@debian:~/prog/shopelectro/docker$ m create-env
# many "y"
andre@debian:~/prog/shopelectro/docker$ dc -f build/docker-compose.yml build --no-cache python-dev
Building python-dev
Step 1/6 : FROM python:slim
 ---> d6f22b3a2b87
Step 2/6 : RUN echo $SRC_DIR && echo 'bugaga'
 ---> Running in 550af64346e3

bugaga
...
```
It seems build accept no env files, even `.env` in context root. And dc doc says, that it is:
https://docs.docker.com/compose/compose-file/compose-file-v2/#env_file

And every our dockerfile doesn't use any var from .env.

So, it seems we can just rm `.env` file with it's creation in bash code.

Reply:
@artemiy312 , this one was not fixed:

> He'll push only one "y" button instead of several

Reply:
@duker33 

>So, it seems we can just rm .env file with it's creation in bash code.

`.env` file is required neither for building images nor substitution in `Dockerfile`, but for `docker-compose.yml` execution. We use substitution in  `docker-compose.yml`

```bash
rm .env
docker-compose up python
WARNING: The VIRTUAL_HOST_EXPOSE_PORT variable is not set. Defaulting to a blank string.
WARNING: The VIRTUAL_HOST_PORT variable is not set. Defaulting to a blank string.
WARNING: The VIRTUAL_HOST_LIVESERVER_PORT variable is not set. Defaulting to a blank string.
WARNING: The SRC_DIR variable is not set. Defaulting to a blank string.
WARNING: The DEPS_DIR variable is not set. Defaulting to a blank string.
WARNING: The REDIS_PASSWORD variable is not set. Defaulting to a blank string.
ERROR: The Compose file './docker-compose.yml' is invalid because:
services.app.ports is invalid: Invalid port "", should be [[remote_ip:]remote_port[-remote_port]:]port[/protocol]
services.app.ports is invalid: Invalid port ":", should be [[remote_ip:]remote_port[-remote_port]:]port[/protocol]
``` 

>this one was not fixed:

yes, because I still do not push any changes

Reply:
@artemiy312 

> but for docker-compose.yml execution

but `dc build` works fine. And it's single purpose of build/docker-compose.yml file, as i see.
It should not up any containers

> yes, because I still do not push any changes

i thought, i commented about it at the first review, sorry.

Reply:
@duker33 

`.env` file is required for `docker/docker-compose*.yml` files

Reply:
@duker33 You can try to execute `docker-compose up`  without `.env` and you will see a such error: https://github.com/fidals/shopelectro/pull/280#discussion_r186714135.

As I said before, the `.env` is required for variable substitution in `docker/docker-compose*.yaml`

e.g.

```yaml
  app:
    image: fidals/se:prod
    container_name: se-python
    restart: always
    env_file:
      - ...
    environment:
      - ...
    depends_on:
      - ...
    ports:
      - $VIRTUAL_HOST_PORT  # it can only be taken from .env
```

Reply:
@artemiy312 , tested manually. You are right.
Here is good explanation
https://github.com/docker/compose/issues/4223#issuecomment-280077263

Reply:
fixed